### PR TITLE
Add Datasources#resourceSchema to the API

### PR DESCRIPTION
**Pull request by wemrysi**
Provides the ability to obtain the schema of a resource and includes an implementation based on SSTs. Exposes this functionality in the REPL via the new `schema` command.

### CODE REVIEW

File: impl/src/main/scala/quasar/impl/schema/package.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import matryoshka.implicits._
+import scalaz._, Scalaz._
+import scalaz.syntax.tag._
+import spire.algebra.Field
+import spire.math.ConvertableTo
+
+package object schema {
+
+  /** Reduces the input to an `SST` summarizing its structure. */
+  def extractSst[F[_], J: Order, A: ConvertableTo: Field: Order](
```
Comment (wemrysi) [inline impl/src/main/scala/quasar/impl/schema/package.scala]: Resurrected verbatim, modulo `Process` => `Stream` conversion, from the old `interface` project.

File: impl/src/test/scala/quasar/impl/schema/ExtractSstSpec.scala [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+import eu.timepit.refined.auto._
+import fs2.Stream
+import matryoshka.data.Fix
+import matryoshka.implicits._
+import scalaz._, Scalaz._
+import spire.std.double._
+import spire.math.Real
+
+final class ExtractSstSpec extends quasar.Qspec {
```
Comment (wemrysi) [inline impl/src/test/scala/quasar/impl/schema/ExtractSstSpec.scala]: Also resurrected from `interface`.

File: impl/src/test/scala/quasar/impl/datasources/DatasourceManagementSpec.scala [hunk 3]
```diff
@@ -40,17 +47,21 @@ import eu.timepit.refined.auto._
...
 import matryoshka.data.Fix
+import matryoshka.implicits._
 import scalaz.{\/, -\/, IMap, Show}
+import scalaz.syntax.bind._
 import scalaz.syntax.either._
 import scalaz.syntax.show._
 import scalaz.std.anyVal._
 import scalaz.std.option._
-import shims._
+import shims.{orderToScalaz => _, eqToScalaz => _, _}
```
Comment (djspiewak) [inline impl/src/test/scala/quasar/impl/datasources/DatasourceManagementSpec.scala]: What conflicts do these create?

File: impl/src/test/scala/quasar/impl/datasources/DatasourceManagementSpec.scala [hunk 4]
```diff
@@ -319,7 +356,7 @@ object DatasourceManagementSpec {
       def kind = kind0
 
       def evaluate(query: Q): F[Stream[F, Data]] =
-        Stream.empty.covaryAll[F, Data].pure[F]
+        Stream.emits(List(true, true, false, true, false) map (Data._bool(_))).covary[F].pure[F]
```
Comment (djspiewak) [inline impl/src/test/scala/quasar/impl/datasources/DatasourceManagementSpec.scala]: ⛳️ `Stream(true, true, false, true, false).map(Data._bool(_)).covary[F].pure[F]`

File: repl/src/main/scala/quasar/repl/Main.scala [hunk 5]
```diff
@@ -46,7 +47,7 @@ object Main extends StreamApp[IO] {
       _ <- Paths.mkdirs[Stream[IO, ?]](dataDir)
       pluginDir = basePath.resolve(Paths.QuasarPluginsDirName)
       _ <- Paths.mkdirs[Stream[IO, ?]](pluginDir)
-      q <- Quasar[IO](dataDir, ExternalConfig.PluginDirectory(pluginDir), global)
+      q <- Quasar[IO](dataDir, ExternalConfig.PluginDirectory(pluginDir), 1000L)
```
Comment (djspiewak) [inline repl/src/main/scala/quasar/repl/Main.scala]: Magic number is magic.

Comment (wemrysi) [reply repl/src/main/scala/quasar/repl/Main.scala]: :sparkles: , yeah, we need to loop back and expose configuration of the repl.

**Pull request merged**